### PR TITLE
Felix branch

### PR DESCRIPTION
**Pull request by Zneed99**
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] I have followed the [perfect PR recommendations][perfect-pr]
- [ ] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

### CODE REVIEW

File: homeassistant/components/huawei_lte/__init__.py [hunk 1]
```diff
@@ -103,6 +103,7 @@
         }
     ),
 )
+# Test comment
```
Comment (OneZeroWasTaken) [inline homeassistant/components/huawei_lte/__init__.py]: Fine but should be removed

File: homeassistant/components/huawei_lte/__init__.py [hunk 2]
```diff
@@ -492,72 +510,68 @@ async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
...
-                (router for router in routers.values() if router.url == url), None
-            )
+        url = service.data.get(CONF_URL)
+        router = None
+
+        if url:
+            router = next((r for r in routers.values() if r.url == url), None)
         elif not routers:
             _LOGGER.error("%s: no routers configured", service.service)
-            return
```
Comment (OneZeroWasTaken) [inline homeassistant/components/huawei_lte/__init__.py:512]: Removing this return seems like changed behavior, right?

File: homeassistant/components/huawei_lte/__init__.py [hunk 3]
```diff
@@ -492,72 +510,68 @@ async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
...
         elif len(routers) == 1:
             router = next(iter(routers.values()))
         else:
             _LOGGER.error(
                 "%s: more than one router configured, must specify one of URLs %s",
                 service.service,
-                sorted(router.url for router in routers.values()),
+                sorted(r.url for r in routers.values()),
             )
-            return
```
Comment (OneZeroWasTaken) [inline homeassistant/components/huawei_lte/__init__.py:521]: Same with this return.
Variable routers is not used after this point either, so I think these returns should be added back

File: homeassistant/components/huawei_lte/__init__.py [hunk 4]
```diff
@@ -492,72 +510,68 @@ async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
...
-            # Login will be handled automatically on demand
-            router.suspended = False
-            _LOGGER.debug("%s: %s", service.service, "done")
-        elif service.service == SERVICE_SUSPEND_INTEGRATION:
-            router.logout()
-            router.suspended = True
-            _LOGGER.debug("%s: %s", service.service, "done")
-        else:
-            _LOGGER.error("%s: unsupported service", service.service)
+        if router.suspended:
```
Comment (OneZeroWasTaken) [inline homeassistant/components/huawei_lte/__init__.py]: Should also check "service.service == SERVICE_CLEAR_TRAFFIC_STATISTICS or service.service == SERVICE_REBOOT " as it did before refactoring

**Review by OneZeroWasTaken — CHANGES_REQUESTED** [review]

File: homeassistant/components/huawei_lte/__init__.py (hunk 4 above)
Comment (Zneed99) [reply homeassistant/components/huawei_lte/__init__.py:539]: Think you missed this one, it is found at line 558

**Review by OneZeroWasTaken — APPROVED** [review]

**Pull request merged**